barretenberg/cpp/src/barretenberg/stdlib/encryption/ecdsa/ecdsa_impl.hpp

```diff
@@ -43,158 +100,119 @@ auto& engine = numeric::get_debug_randomness();
-    result_mod_r.binary_basis_limbs[1].maximum_value = result.x.binary_basis_limbs[1].maximum_value;
-    result_mod_r.binary_basis_limbs[2].maximum_value = result.x.binary_basis_limbs[2].maximum_value;
-    result_mod_r.binary_basis_limbs[3].maximum_value = result.x.binary_basis_limbs[3].maximum_value;
-
-    result_mod_r.prime_basis_limb = result.x.prime_basis_limb;
-
-    result_mod_r.assert_is_in_field();
-
-    result_mod_r.binary_basis_limbs[0].element.assert_equal(r.binary_basis_limbs[0].element);
-    result_mod_r.binary_basis_limbs[1].element.assert_equal(r.binary_basis_limbs[1].element);
-    result_mod_r.binary_basis_limbs[2].element.assert_equal(r.binary_basis_limbs[2].element);
-    result_mod_r.binary_basis_limbs[3].element.assert_equal(r.binary_basis_limbs[3].element);
-    result_mod_r.prime_basis_limb.assert_equal(r.prime_basis_limb);
-    return bool_t<Builder>(ctx, true);
-}
 
-/**
- * @brief Verify ECDSA signature. Returns 0 if signature fails (i.e. does not produce unsatisfiable constraints)
- *
- * @tparam Builder
- * @tparam Curve
- * @tparam Fq
- * @tparam Fr
- * @tparam G1
- * @param hashed_message
- * @param public_key
- * @param sig
- * @return bool_t<Builder>
- */
-template <typename Builder, typename Curve, typename Fq, typename Fr, typename G1>
-bool_t<Builder> ecdsa_verify_signature_prehashed_message_noassert(const stdlib::byte_array<Builder>& hashed_message,
-                                                                  const G1& public_key,
-                                                                  const ecdsa_signature<Builder>& sig)
-{
-    Builder* ctx = hashed_message.get_context() ? hashed_message.get_context() : public_key.x.context;
-
-    Fr z(hashed_message);
-    z.assert_is_in_field();
-
-    Fr r(sig.r);
-    // force r to be < secp256k1 group modulus, so we can compare with `result_mod_r` below
-    r.assert_is_in_field();
-
-    Fr s(sig.s);
-
-    // r and s should not be zero
-    r.assert_is_not_equal(Fr::zero());
-    s.assert_is_not_equal(Fr::zero());
+    // Step 5.
+    if (result.get_value().is_point_at_infinity()) {
+        info("The result of the batch multiplication is the point at infinity. This will produce an unsatisfied "
+             "circuit.");
+    }
+    result.is_point_at_infinity().assert_equal(bool_t<Builder>(false));
 
-    // s should be less than |Fr| / 2
-    // Read more about this at: https://www.derpturkey.com/inherent-malleability-of-ecdsa-signatures/amp/
-    s.assert_less_than((Fr::modulus + 1) / 2);
+    // Step 6.
+    // We reduce result.x to 2^s, where s is the smallest s.t. 2^s > q. It is cheap in terms of constraints, and avoids
+    // possible edge cases
+    result.x.self_reduce();
 
-    Fr u1 = z / s;
-    Fr u2 = r / s;
+    // Transfer Fq value result.x to Fr (this is just moving from a C++ class to another)
+    Fr result_x_mod_r = Fr::unsafe_construct_from_limbs(result.x.binary_basis_limbs[0].element,
```

Comment:
I think this changes the `maximum_value` of the limbs - it now sets them to default but previously it was being taken from `result.x`. Can you check if this is correct? If it is, is that a problem?